datafusion/physical-plan/src/joins/hash_join.rs

```diff
@@ -282,7 +284,7 @@ pub struct HashJoinExec {
     pub filter: Option<JoinFilter>,
     /// How the join is performed (`OUTER`, `INNER`, etc)
     pub join_type: JoinType,
-    /// The output schema for the join
+    /// The schema after join
```

Comment:
- Add projection
- update `equivalence_properties` and `output_ordering` after projection
- update `column_indices`
- keep `HashJoinExec.schema` as the result of `build_join_schema` , we can get the finally schema after projection through `schema()` function.  So we need to be careful when using it.